packages/shell-api/src/shell-internal-state.ts

```diff
@@ -0,0 +1,238 @@
+   * properties but also have the global shell api objects and functions.
+   *
+   * @param {Object} contextObject - contextObject an object used as global context.
+   */
+  setCtx(contextObject: any): void {
+    // Add API methods for VSCode and scripts
+    contextObject.use = this.use.bind(this);
+    contextObject.show = this.show.bind(this);
+    contextObject.it = this.it.bind(this);
+    contextObject.help = this.help.bind(this);
+    contextObject.toIterator = toIterator;
+    contextObject.print = async(arg) => {
+      if (arg.toReplString) {
+        console.log(await arg.toReplString());
+      } else {
+        console.log(arg);
+      }
+    };
+    contextObject.printjson = contextObject.print;
+    Object.assign(contextObject, ShellBson);
+    contextObject.rs = new ReplicaSet(this.currentDb.mongo);
+    contextObject.sh = new Shard(this.currentDb.mongo);
+
+    // Add global shell objects
+    this.asyncWriter.symbols.initializeApiObjects({
+      db: signatures.Database,
+      rs: signatures.ReplicaSet,
+      sh: signatures.Shard,
+      Mongo: signatures.Mongo
+    });
+
+    try {
+      Object.defineProperty(contextObject, 'db', {
+        set: (newDb: any) => {
+          if (newDb.shellApiType && newDb.shellApiType() !== 'Database') {
+            throw new MongoshInvalidInputError('Cannot reassign \'db\' to non-Database type');
+          }
+          this.currentDb = newDb;
+          contextObject.rs = new ReplicaSet(this.currentDb.mongo);
+          contextObject.sh = new Shard(this.currentDb.mongo);
+          this.fetchConnectionInfo();
+          return newDb;
+        },
+        get: () => {
+          return this.currentDb;
+        }
+      });
+    } catch (e) { // java shell, can't use getters/setters
+      console.log(`in java shell, platform: ${this.initialServiceProvider.platform}`);
+      contextObject.db = this.currentDb;
+    }
+
+    contextObject.Mongo = async(uri?, options?): Promise<Mongo> => {
+      if (
+        this.initialServiceProvider.platform !== ReplPlatform.CLI
+      ) {
+        throw new MongoshUnimplementedError(
+          `new Mongo connection are not supported for current platform: ${ReplPlatform[this.initialServiceProvider.platform]}`
+        );
+      }
+      const mongo = new Mongo(this, uri, options);
+      await mongo.connect();
+      return mongo;
+    };
+
+    this.messageBus.emit(
+      'mongosh:setCtx',
+      { method: 'setCtx', arguments: { db: contextObject.db } }
+    );
+  }
+
+  public emitApiCall(event: {
+    method: string;
+    class: string;
+    arguments: Document;
+    [otherProps: string]: any;
+  }): void {
+    this.messageBus.emit('mongosh:api-call', event);
+  }
+
+  /**
+   * Helper method to adapt aggregation pipeline options.
+   * This is here so that it's not visible to the user.
+   *
+   * @param options
+   */
+  public adaptAggregateOptions(options: any = {}): {
```

Comment:
I'd look for a way to move this one to a util class or just as a separate function in a module, looks like completely static.

Reply:
I moved it into a `helper.ts` file that is stateless, good idea!